internal/pkg/slicer/config/config.go

```diff
@@ -11,28 +17,71 @@ const (
 
 type Mode uint
 
+type ByteSize = datasize.ByteSize
+
 type Config struct {
-	Mode             Mode   `json:"mode" validate:"required"`
-	BytesPerSlice    uint64 `json:"bytesPerSlice" validate:"min=1"`
-	RowsPerSlice     uint64 `json:"rowsPerSlice" validate:"min=1"`
-	NumberOfSlices   uint32 `json:"numberOfSlices" validate:"min=1"`
-	MinBytesPerSlice uint64 `json:"minBytesPerSlice" validate:"min=1"` // if Mode = ModeSlices
-	Gzip             bool   `json:"gzip"`
-	GzipLevel        int    `json:"gzipLevel" validate:"min=1,max=9"`
+	Mode             Mode              `json:"mode" validate:"required"`
+	BytesPerSlice    datasize.ByteSize `json:"bytesPerSlice" validate:"min=1"`
+	RowsPerSlice     uint64            `json:"rowsPerSlice" validate:"min=1"`
+	NumberOfSlices   uint32            `json:"numberOfSlices" validate:"min=1"`
+	MinBytesPerSlice datasize.ByteSize `json:"minBytesPerSlice" validate:"min=1"` // if Mode = ModeSlices
+	Gzip             bool              `json:"gzip"`
+	GzipLevel        int               `json:"gzipLevel" validate:"min=1,max=9"`
 }
 
 func DefaultConfig() Config {
 	return Config{
 		Mode:             ModeBytes,
-		BytesPerSlice:    524_288_000, // 500 MiB
+		BytesPerSlice:    500 * datasize.MB,
 		RowsPerSlice:     1_000_000,
 		NumberOfSlices:   60,
-		MinBytesPerSlice: 4194304, // 4 MiB
+		MinBytesPerSlice: 4 * datasize.MB,
 		Gzip:             true,
 		GzipLevel:        2, // 1 - BestSpeed, 9 - BestCompression
 	}
 }
 
+func (v *Config) UnmarshalJSON(data []byte) error {
+	// Decode to a map
+	m := make(map[string]any)
+	err := json.Unmarshal(data, &m)
+	if err != nil {
+		return err
+	}
+
+	// Convert bytes as int, to bytes as string, for example 123 -> "123B".
+	// This ensures backward compatibility with older configurations.
+	if i, ok := m["bytesPerSlice"].(float64); ok {
+		m["bytesPerSlice"] = strconv.FormatInt(int64(i), 10) + "B"
+	}
+	if i, ok := m["minBytesPerSlice"].(float64); ok {
+		m["minBytesPerSlice"] = strconv.FormatInt(int64(i), 10) + "B"
+	}
+
+	// Encode update version
+	data, err = json.Marshal(m)
+	if err != nil {
+		return err
+	}
+
+	// Decode to the struct, skip UnmarshalJSON implementation
+	type _c Config
+	return json.Unmarshal(data, (*_c)(v))
+}
```

Comment:
`datasize` lib-ka nepodporuje `unmarshal` z `int/number`, preto tento kod.
Konfig sa najprv dekoduje ako `map[string]any`, upravia sa tieto hodnoty, prevedie sa to spat na JSON, a dekoduje sa to do struktury.